lib/state_machines/machine.rb

```diff
@@ -2046,14 +2046,12 @@ def action_hook
     # the method and is further along in the ancestor chain than this
     # machine's helper module.
     def owner_class_ancestor_has_method?(scope, method)
+      return false unless owner_class_has_method?(scope, method)
+
       superclasses = owner_class.ancestors[1..-1].select { |ancestor| ancestor.is_a?(Class) }
 
       if scope == :class
-        # Use singleton classes
-        current = (
-        class << owner_class;
-          self;
-        end)
+        current = owner_class.singleton_class
```

Comment:
`singleton_class` was introduced with MRI 1.9.2, I think it's safe to use it now.